core/java/iesi-core/src/main/java/io/metadew/iesi/script/execution/instruction/data/text/TextSubstring.java

```diff
@@ -27,16 +27,23 @@ public String generateOutput(String parameters) {
             String text = inputParameterMatcher.group(FIRST_OPERATOR);
             int start = Integer.parseInt(inputParameterMatcher.group(SECOND_OPERATOR))-1;
             int end = Integer.parseInt(inputParameterMatcher.group(THIRD_OPERATOR));
-
             //if the user put a negative value, add length() and +1=> in order to take the 0 into account
             if (start < 0)
                 start += text.length()+1;
 
             if (end < 0)
                 end += text.length()+1 ;
 
-            verifyArguments(text, start, end);
-            return text.substring(start, end);
+            //Those lines must be after checking negative values
+            int counterNewLinesBefore = countNumberOfLines(text,end,inputParameterMatcher.groupCount());
+            if(counterNewLinesBefore!=0){
+                start+= counterNewLinesBefore;
+                end+= counterNewLinesBefore;
+            }
+            int counterNewLinesInside = countNumberOfLinesInside(text,start,end);
+
+            verifyArguments(text, start-counterNewLinesInside, end);
+            return text.substring(start-counterNewLinesInside, end);
```

Comment:
formatting of code. In IntelliJ you can auto format the code. Normally SonarLint should also complain about this formatting. Is the plugin installed at your side?